### DISCUSSION TITLE
Possibility to configure TTL for file system

### DISCUSSION THREAD

**Discussion by MauriceArikoglu** [caching-tools/next-shared-cache#290] [General]
When using a cache handler with `useFileSystem` set to `true` the handler falls back to serving content from the disk. Is there a possibility to set a TTL for file system data? How can I control the duration of serving from cache and still use the file system cache (we need to because of pages router https://github.com/vercel/next.js/issues/58094)

**Comment by better-salmon** [comment]
Hi @MauriceArikoglu! I've been thinking about this feature since the beginning of the year. It's a complicated task because the revalidation period information is not stored in cache values that are saved in the file system. However, this feature is vital and will be the next big thing before we release version 1.0.0.

It is recommended to avoid using TTL features with Pages Router. Instead, use [the Key Eviction](https://redis.io/docs/reference/eviction/) feature if Redis is used as a cache store. The LRU strategy is the recommended approach.

One hack could help, but I must test it first. Please stay tuned for updates on this discussion.

Thank you very much for using the library and providing feedback!

**Reply by MauriceArikoglu** [reply]
Are you available for discussion on this topic via discord / slack / call?

**Reply by better-salmon** [reply]
Maybe tomorrow (28.01.2024) via Discord call? But I must warn you that Grammarly hasn't yet created an app for the human brain. I speak English slowly and sometimes forget or misuse words 😆

**Reply by better-salmon** [reply]
I've created a Discord server https://discord.gg/ZJtHDaXD. @MauriceArikoglu you are welcome!

**Comment by better-salmon** [comment]
@MauriceArikoglu I've found that you can use arbitrary `revalidate` values in Pages Router's `getStaticProps` function.

This code works:

```js
export async function getStaticProps() {
    const result = await fetch('...');

    const parsedResult = await result.json();

    return {
        props: {
            // ...
        },
        revalidate: parsedResult.revalidateValueFromBackend,
    };
}
```

Even if it is not a common setup, we can not fully rely on parsing source files with the `getStaticProps` function.

**Reply by MauriceArikoglu** [reply]
@better-salmon I've tried to come up with a scenario where this is used but couldn't find some.

I propose to solve this by allowing to configure a "fallback" revalidation period and logging explicitly where the revalidation value comes from ("using parsed revalidation", "using fallback revalidation"...).

Also I would offer to write some documentation that explains the feature and it's caveats. What do you think?

**Reply by MauriceArikoglu** [reply]
@better-salmon actually, looking at the code in #313 we are parsing the values into the Manifest. We could show the result to the user and even allow providing fallback for routes where no value was found.

I would propose the simple solution first and see if the use case really exists.
